### PR TITLE
Add pd-workshop filter UI

### PR DESCRIPTION
**Pull request by aoby**
Now, on the main workshop index page, the Upcoming and Ended workshops are capped at 5 and an `N more...` button is displayed when more exist:
image

Clicking that goes to the filter UI:
image

There the results can be filtered by status, scheduled dates, course, subject (for relevant courses), and organizer.

Sorting columns on the workshop tables on both pages now retrieve a new sorted result set from the API when fewer than all relevant workshops are loaded. In cases where all workshops for a given query are already loaded, sorting happens on the client.

There is also a new Filter View button on the main page:
image

Note, this depends on https://github.com/code-dot-org/code-dot-org/pull/13663

### CODE REVIEW

File: apps/src/code-studio/pd/workshop_dashboard/workshop_dashboard.jsx [hunk 1]
```diff
@@ -40,6 +41,11 @@ const WorkshopDashboard = (
         component={WorkshopIndex}
       />
       <Route
+        path="workshops/filter"
+        breadcrumbs="Workshops,Filter"
+        component={WorkshopFiler}
```
Comment (mehalshah) [inline apps/src/code-studio/pd/workshop_dashboard/workshop_dashboard.jsx]: Typo - should be WorkshopFilter

File: apps/src/code-studio/pd/workshop_dashboard/workshop_filter.jsx [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+    }
+  },
+
+  componentDidMount() {
+    this.organizersLoadRequest = $.ajax({
+      method: 'GET',
+      url: '/api/v1/pd/workshop_organizers',
+      dataType: 'json'
+    })
+    .done(data => {
```
Comment (mehalshah) [inline apps/src/code-studio/pd/workshop_dashboard/workshop_filter.jsx:86]: Handle error here - if nothing else, log it to the console and pop up a "We're Sorry" dialog

File: apps/src/code-studio/pd/workshop_dashboard/workshop_filter.jsx [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+        </Row>
+        <Row>
+          <Col sm={4}>
+            <FormGroup>
+              <ControlLabel>Course</ControlLabel>
+              <Select
+                value={filters.course}
+                onChange={this.handleCourseChange}
+                placeholder={null}
+                options={window.dashboard.workshop.COURSES.map(v => ({value: v, label: v}))}
```
Comment (mehalshah) [inline apps/src/code-studio/pd/workshop_dashboard/workshop_filter.jsx:299]: Consider either alphabetizing courses or making the dropdown size larger to include everything

File: apps/src/code-studio/pd/workshop_dashboard/workshop_index.jsx [hunk 4]
```diff
@@ -7,6 +7,9 @@ import {Button, ButtonToolbar} from 'react-bootstrap';
 import WorkshopTable from './components/workshop_table';
 import WorkshopTableLoader from './components/workshop_table_loader';
 
+const MAX_WORKSHOPS_PER_TABLE = 10;
```
Comment (mehalshah) [inline apps/src/code-studio/pd/workshop_dashboard/workshop_index.jsx]: 10 isn't that much - suggest making this larger

Comment (aoby) [reply apps/src/code-studio/pd/workshop_dashboard/workshop_index.jsx]: Actually the feedback at Wed's bug bash was to reduce it to 5. See https://trello.com/c/bW0cQas5/20-add-workshop-filters

File: apps/src/code-studio/pd/workshop_dashboard/workshop_dashboard.jsx (hunk 1 above)
Comment (aoby) [reply apps/src/code-studio/pd/workshop_dashboard/workshop_dashboard.jsx]: What if I want to file a workshop :trollface: ?

File: apps/src/code-studio/pd/workshop_dashboard/components/server_sort_workshop_table.jsx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    if (this.workshops && this.props.generateCaptionFromWorkshops) {
+      return this.props.generateCaptionFromWorkshops(this.workshops);
+    }
+  },
+
+  handleTableSort(sort) {
+    const limit = this.workshops.limit;
+    const workshopCount = this.workshops.total_count;
+
+    // As an optimization, only re-query with an oderBy when fewer than all workshops are loaded.
```
Comment (mehalshah) [inline apps/src/code-studio/pd/workshop_dashboard/components/server_sort_workshop_table.jsx]: order

File: apps/src/code-studio/pd/workshop_dashboard/components/workshop_table_loader.jsx [hunk 6]
```diff
@@ -9,14 +9,16 @@
...
 import Spinner from '../components/spinner';
 
 const WorkshopTableLoader = React.createClass({
   propTypes: {
     queryUrl: React.PropTypes.string.isRequired,
     canDelete: React.PropTypes.bool, // When true, sets child prop onDelete to this.handleDelete
     children: React.PropTypes.element.isRequired, // Require exactly 1 child component.
-    hideNoWorkshopsMessage: React.PropTypes.bool // Should we show "no workshops found" if no workshops are found?
+    hideNoWorkshopsMessage: React.PropTypes.bool, // Should we show "no workshops found" if no workshops are found?
+    params: React.PropTypes.object
```
Comment (mehalshah) [inline apps/src/code-studio/pd/workshop_dashboard/components/workshop_table_loader.jsx]: It's not clear what params are based on this name

File: apps/src/code-studio/pd/workshop_dashboard/workshop_filter.jsx [hunk 7]
```diff
@@ -0,0 +1,352 @@
...
+      dataType: 'json'
+    })
+    .done(data => {
+      this.setState({
+        organizersLoading: false,
+        organizers: data
+      });
+    })
+    .fail((data) => {
+      if (data.statusText !== "abort") {
```
Comment (mehalshah) [inline apps/src/code-studio/pd/workshop_dashboard/workshop_filter.jsx:93]: Log the error to the console

File: apps/src/code-studio/pd/workshop_dashboard/components/workshop_table_loader.jsx (hunk 6 above)
Comment (aoby) [reply apps/src/code-studio/pd/workshop_dashboard/components/workshop_table_loader.jsx]: good point. How about queryParams?

**Pull request merged**